src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final String INDEX_ERROR_MESSAGE = "Array list index out of bounds";
+    private static final String ELEMENT_ERROR_MESSAGE = "No such element in list";
+    private int size = 0;
+    private int capacity = 10;
+    private Object[] listArr;
+
+    public ArrayList() {
+        listArr = new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= capacity) {
+            extendCapacity();
+        }
+        listArr[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_ERROR_MESSAGE);
+        }
+        if (size >= capacity) {
+            extendCapacity();
+        }
```

Comment:
code duplication with 18-20 lines, can me a separate method and called extendIfNecessary or like that